### PR TITLE
[SEE https://github.com/credo-ai/credoai_lens/pull/270 FIRST] Feat/test coverage

### PR DESCRIPTION
**Pull request by esherman-credo**
## Describe your changes
Added test cases to improve coverage.
Added support for ModelProfiler and Shap evaluators
Added some validation tests

Currently fails due to revealed bugs/inadequate input handling

## Issue ticket number and link
See: https://github.com/credo-ai/credoai_lens/pull/270
This branch has been rebased to the above. We should merge the above first, then this one (after bug fixes...separate branch).

## Known outstanding issues that are not fully accounted for
Outstanding bugs/validation issues:

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have built basic tests for new functionality (particularly new evaluators)
- N/A
- [x] If new libraries have been added, I have checked that [readthedocs](https://readthedocs.org/projects/credoai-lens/) API documentation is constructed correctly
- N/A
- [x] Will this be part of a major product update? If yes, please write one phrase about this update.
- No

## Extra-mile Checklist
- [x] I have thought expansively about edge cases and written tests for them

### CODE REVIEW

**Comment by esherman-credo** [comment]
See #274 for item 4 in the above list of outstanding issues.

**Comment by esherman-credo** [comment]
See #276 for fix to item 5 in above list of outstanding issues.

File: tests/test_lens.py [hunk 1]
```diff
@@ -153,6 +167,28 @@ def test_identity_verification(init_lens_identityverification):
     pytest.assume(results_fair.equals(expected_results["fair"]))
 
 
+@pytest.mark.parametrize(
```
Comment (fabrizio-credo) [inline tests/test_lens.py:170]: Very cool!

File: tests/test_lens.py [hunk 2]
```diff
@@ -179,3 +215,26 @@ def test_empty_pipeline_run(
         training_data=classification_train_data,
     )
     my_pipeline.run()
+
+
+def test_lens_validation_no_sens_feat(init_lens_classification):
```
Comment (fabrizio-credo) [inline tests/test_lens.py:220]: This could be simplified like:

```python
@pytest.mark.xfail(raises=ValidationError)
def test_lens_validation_no_sens_feat(init_lens_classification):
    """
    Tests to ensure Lens will not allow running evaluators that require sensitive features without
    any sensitive features specified
    """
    lens, _, _ = init_lens_classification
    lens.assessment_data.sensitive_features = None
    lens.sens_feat_names = []
    lens.training_data = None
    lens.add(ModelFairness(["accuracy_score"])
```
There are also DataFairness, equity, and RankingFairness that have the same requirements. Can you parametrize if possible, or spawn new tests?

File: credoai/evaluators/identity_verification.py [hunk 3]
```diff
@@ -161,6 +161,12 @@ def _setup(self):
 
         return self
 
+    def _validate_arguments(self):
+        check_data_instance(self.assessment_data, ComparisonData)
+        check_model_instance(self.model, (ComparisonModel, DummyComparisonModel))
+        check_existence(self.assessment_data.pairs, "pairs")
+        return self
+
```
Comment (fabrizio-credo) [inline credoai/evaluators/identity_verification.py:169]: The entire method is duplicated, please retain only the version above _setup().

**Review by fabrizio-credo — CHANGES_REQUESTED** [review]
Excellent additions, a few changes required to speed up/simplify tests.

File: tests/test_artifacts.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    """
+    tests call to Base.data property
+    """
+    X_test, y_test, _, _ = fetch_credit_model()
+
+    credo_data = TabularData(name="test_data_property", X=X_test, y=y_test)
+    pytest.assume(not print(credo_data.data))
+
+
+def test_tabular_data_array_inputs():
```
[the remark this answers is not in the document]
Comment (esherman-credo) [reply tests/test_artifacts.py:33]: This will be fixed by #277

File: credoai/evaluators/identity_verification.py (hunk 3 above)
Comment (esherman-credo) [reply credoai/evaluators/identity_verification.py:169]: Must've been something that came up in merge...?

File: tests/test_artifacts.py [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+
+def test_base_data_property():
+    """
+    tests call to Base.data property
+    """
+    X_test, y_test, _, _ = fetch_credit_model()
+
+    credo_data = TabularData(name="test_data_property", X=X_test, y=y_test)
+    pytest.assume(not print(credo_data.data))
```
[the remark this answers is not in the document]
Comment (esherman-credo) [reply tests/test_artifacts.py:27]: The point is explicitly to cover the `.data` property of BaseData (through TabularData).

It's a not because the print function doesn't return anything. I could just do `pytest.assume(credo_data.data)`

File: tests/test_artifacts.py [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+    X = X.drop("MARRIAGE", axis=1)
+
+    credo_data = TabularData(
+        "test_sens_feat_interactions",
+        X=X,
+        y=y,
+        sensitive_features=sens,
+        sensitive_intersections=True,
+    )
+    test_intersections = credo_data.sensitive_features
```
[the remark this answers is not in the document]
Comment (esherman-credo) [reply tests/test_artifacts.py:68]: "An xfail means that you expect a test to fail for some reason. A common example is a test for a feature not yet implemented, or a bug not yet fixed. When a test passes despite being expected to fail (marked with pytest.mark.xfail), it’s an xpass and will be reported in the test summary."

https://docs.pytest.org/en/7.1.x/how-to/skipping.html

That's not really what's going on here. I'm asserting that the `sensitive_features` should've changed relative to the input passed to the constructor.

The point of using the `with ... e_info` is becuase the `e_info` structure will contain some info if the error doesn't get thrown or if it's an error that isn't the expected AssertionError

**Review by fabrizio-credo — APPROVED** [review]
